wal/repair.go

```diff
@@ -32,15 +32,20 @@ func Repair(dirpath string) bool {
 	}
 	defer f.Close()
 
-	n := 0
+	n := int64(0)
 	rec := &walpb.Record{}
 
 	decoder := newDecoder(f)
 	for {
+		lastOff := decoder.lastOffset()
 		err := decoder.decode(rec)
 		switch err {
 		case nil:
-			n += 8 + rec.Size()
+			n += decoder.lastOffset() - lastOff
+			if lastOff > decoder.lastOffset() {
```

Comment:
see line29, this should only open last file for reading? so this should not happen?